libs/core/src/test/java/org/elasticsearch/common/util/ESSloppyMathTests.java

```diff
@@ -42,4 +45,14 @@ public void testSinh() {
             assertEquals(StrictMath.sinh(d), sinh(d), SINH_DELTA);
         }
     }
+
+    public void testLog() {
+        assertTrue(Double.isNaN(log(Double.NaN)));
+        assertEquals(Double.NaN, log(Double.NEGATIVE_INFINITY), LOG_DELTA);
+        assertEquals(Double.POSITIVE_INFINITY, log(Double.POSITIVE_INFINITY), LOG_DELTA);
+        for (int i = 0; i < 10000; i++) {
+            double d = randomDoubleBetween(-Double.MAX_VALUE, Double.MAX_VALUE, true);
```

Comment:
Since the equations above seem to have a bunch of special cases between 0 and 0.95, and 0.95 and 1.14, I think the current test is almost never going to touch those values.

I suggest either a few more explicit tests for values in those ranges, or add an additional test here with random between 0 and 2, perhaps?

Reply:
OK. I added some more testing in https://github.com/elastic/elasticsearch/pull/87515/commits/23d7005e6445341d53e776bc7246290cd80fe655